src/rest.cpp

```diff
@@ -254,11 +254,11 @@ static bool rest_block(HTTPRequest* req,
 
         if (IsBlockPruned(pblockindex))
             return RESTERR(req, HTTP_NOT_FOUND, hashStr + " not available (pruned data)");
-
-        if (!ReadBlockFromDisk(block, pblockindex, Params().GetConsensus()))
-            return RESTERR(req, HTTP_NOT_FOUND, hashStr + " not found");
     }
 
+    if (!ReadBlockFromDisk(block, pblockindex, Params().GetConsensus()))
```

Comment:
You need to call with `CDiskBlockPos`, not with `CBlockIndex*`.

Reply:
I only moved the two lines out of the lock scope, the overload for `ReadBlockFromDisk` does that (I think `CDiskBlockPos` was renamed to `FlatFilePos` in 65a489e93d181d3c0f7a9cf79f7c11ff8cf2b0f0)

Reply:
(While touching this) since the Apple "goto fail" vulnerability (https://dwheeler.com/essays/apple-goto-fail), we add brackets for conditionals of more than one line

```diff
-    if (!ReadBlockFromDisk(block, pblockindex, Params().GetConsensus()))
+    if (!ReadBlockFromDisk(block, pblockindex, Params().GetConsensus())) {
         return RESTERR(req, HTTP_NOT_FOUND, hashStr + " not found");
+    }
```